### PR TITLE
Feature/29 azure pipeline

### PR DESCRIPTION
**Pull request by pedro-lima-jalafund**
[user story #29](https://dev.azure.com/funjaladev/DevLevel3/_boards/board/t/DevLevel3%20Team/Stories/?workitem=29​​​​​​​)
[AB#29](https://dev.azure.com/funjaladev/7867a1c5-5ac7-4c7f-baff-87d740bf4fc1/_workitems/edit/29)

### CODE REVIEW

**Comment by pedro-lima-jalafund** [comment]
Still missing if after merge the condition of `Save artifact` works

Knowing about this will be useful in the future
https://github.com/fundev3/teamup-webapp/pull/2#discussion_r670477254

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Pipeline for the TeamUp's webApp
+
+trigger:
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml:1]: delete two previuous lines, let's make this the first line of the yml

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  # batch runs this pipeline one after the other (useful when there are many contributors)
+  batch: true
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: remove batch

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+pool:
+  vmImage: ubuntu-latest
+
+steps:
+- task: NodeTool@0
+  inputs:
+    versionSpec: '14.17.x' 
+  displayName: 'Installi Node 14.17.x'
+
+# installing dependencies (react, axios, etc)
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: delete all comments please, let's keep using meaningful names for all tasks

File: azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+
+- task: PublishBuildArtifacts@1
+  displayName: 'Save artifact'
+  inputs:
+    PathtoPublish: '$(Agent.BuildDirectory)/s/build'
+    ArtifactName: 'drop'
+    publishLocation: 'Container'
+#not save artifact on PRs
+  condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: fix indent

File: azure-pipelines.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+# installing dependencies (react, axios, etc)
+- script: npm ci
+  displayName: 'Install dependencies'
+
+# building the webApp
+- script: npm run build
+  displayName: 'Build React webApp'
+#for conditional run of a step or job or stage
+
+
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml:28]: delete unnecesary blank line

**Review by martin-callejas-jalafund — CHANGES_REQUESTED** [review]
good work, but in some cases I had to make the same observations again :(

File: azure-pipelines.yml (hunk 1 above)
Comment (pedro-lima-jalafund) [reply azure-pipelines.yml:1]: done

File: azure-pipelines.yml (hunk 2 above)
Comment (pedro-lima-jalafund) [reply azure-pipelines.yml]: done

File: azure-pipelines.yml (hunk 3 above)
Comment (pedro-lima-jalafund) [reply azure-pipelines.yml]: done

File: azure-pipelines.yml (hunk 4 above)
Comment (pedro-lima-jalafund) [reply azure-pipelines.yml]: on `condition:...` line? it says me that it is and **unexpected property condition** warning

File: azure-pipelines.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  drafts: false
+
+pool:
+  vmImage: ubuntu-latest
+
+steps:
+- task: NodeTool@0
+  inputs:
+    versionSpec: '14.17.x' 
+  displayName: 'Installi Node 14.17.x'
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: let keep it at "Install Node"

File: azure-pipelines.yml (hunk 5 above)
Comment (martin-callejas-jalafund) [reply azure-pipelines.yml:28]: please

File: azure-pipelines.yml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+- script: npm ci
+  displayName: 'Install dependencies'
+
+- script: npm run build
+  displayName: 'Build React webApp'
+
+- task: PublishBuildArtifacts@1
+  displayName: 'Save artifact'
+  condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml:31]: this says "execute only for PullRequests"

Comment (martin-callejas-jalafund) [reply azure-pipelines.yml:31]: we need somethinkg like `condition: and(succeeded(), eq(variables['Build.SourceBranchName'], 'master'))`, but that includes `develop` branch in addtion to master

Comment (martin-callejas-jalafund) [reply azure-pipelines.yml:31]: TIL: ne stands for not equal

Comment (pedro-lima-jalafund) [reply azure-pipelines.yml:31]: I guess it says **not equal** `ne` for pull requests.

**Review by martin-callejas-jalafund — APPROVED** [review]
Awesome work Pedro

**Pull request merged**